spark/src/main/scala/eval/directive/RDDOpDirectives.scala

```diff
@@ -1,118 +1,257 @@
+        {_.localUnequal(_)}, {(d,r) => r.localUnequal(d)},
+        {_.localUnequal(_)}, res1, res2)
+
+    Valid(results)
+  }
+
+  val lessThan = Directive { case (Less(_), res1 :: res2 :: Nil) =>
+    Valid(reduce({_ < _}, {_ <<: _}, {_ < _}, {_ <<: _}, {_ < _}, res1, res2))
+  }
+
+  val lessThanOrEqualTo = Directive { case (LessOrEqual(_), res1 :: res2 :: Nil) =>
+    Valid(reduce({_ <= _}, {_ <=: _}, {_ <= _}, {_ <=: _}, {_ <= _}, res1, res2))
+  }
+
+  val greaterThan = Directive { case (Greater(_), res1 :: res2 :: Nil) =>
+    Valid(reduce({_ > _}, {_ >>: _}, {_ > _}, {_ >>: _}, {_ > _}, res1, res2))
+  }
+
+  val greaterThanOrEqualTo = Directive { case (GreaterOrEqual(_), res1 :: res2 :: Nil) =>
+    Valid(reduce({_ >= _}, {_ >=: _}, {_ >= _}, {_ >=: _}, {_ >= _}, res1, res2))
+  }
+
+  // TODO The mask can probably be something other than `Polygon`.
+  val masking = Directive {
+    case (Masking(_), RDDResult(r) :: GeomResult(g: Polygon) :: Nil) => Valid(RDDResult(r.mask(g)))
+    case (Masking(_), GeomResult(g: Polygon) :: RDDResult(r) :: Nil) => Valid(RDDResult(r.mask(g)))
+  }
+
+  val atan2 = Directive { case (Atan2(_), childResults) =>
+    val ri: (TileLayerRDD[SpatialKey], Int) => TileLayerRDD[SpatialKey] = { (r, i) =>
+      r.withContext(_.mapValues(_.mapDouble(math.atan2(i.toDouble, _))))
+    }
+
+    val rd: (TileLayerRDD[SpatialKey], Double) => TileLayerRDD[SpatialKey] = { (r, d) =>
+      r.withContext(_.mapValues(_.mapDouble(math.atan2(d, _))))
+    }
+
+    val results: Result = childResults match {
+      case (RDDResult(r1) :: RDDResult(r2) :: Nil) =>
+        RDDResult(binary({ (a,b) => a.combineValues(b) { (t1,t2) => t1.combineDouble(t2)(math.atan2) }}, r1, r2))
+      case (RDDResult(r) :: IntResult(i) :: Nil) => RDDResult(ri(r, i))
+      case (IntResult(i) :: RDDResult(r) :: Nil) => RDDResult(ri(r, i))
+      case (RDDResult(r) :: DoubleResult(d) :: Nil) => RDDResult(rd(r, d))
+      case (DoubleResult(d) :: RDDResult(r) :: Nil) => RDDResult(rd(r, d))
+    }
+
+    Valid(results)
+  }
+
+  val pow = Directive { case (Pow(_), res1 :: res2 :: Nil) =>
+    val f: (RDD[(SpatialKey, Tile)], RDD[(SpatialKey, Tile)]) => RDD[(SpatialKey, Tile)] = { (a,b) =>
+      a.combineValues(b) { (t1,t2) => t1.combineDouble(t2)(math.pow) }
+    }
+
+    Valid(reduce({_ localPow _}, { (i,r) => r.localPow(i) }, { _ localPow _ }, { (d,r) => r.localPow(d) }, f, res1, res2))
+  }
+
+  /* --- UNARY EXPRESSIONS --- */
+
+  val classify = Directive { case (Classification(_, classMap), RDDResult(r) :: Nil) =>
+    unary({ _.color(ColorMap(classMap.classifications)) }, r)
+  }
+
+  val sin = Directive { case (Sin(_), RDDResult(r) :: Nil) => unary({ _.localMapDouble(math.sin(_)) }, r) }
+  val cos = Directive { case (Cos(_), RDDResult(r) :: Nil) => unary({ _.localMapDouble(math.cos(_)) }, r) }
+  val tan = Directive { case (Tan(_), RDDResult(r) :: Nil) => unary({ _.localMapDouble(math.tan(_)) }, r) }
```

Comment:
Do these families of functions have to account for the case where the input isn't an RDD? Do we allow these basic mathy ops on Int/Double?

Reply:
Yeah, definitely - it is less pressing for the `RDD` interpreter I suppose - but (e.g.) the local case supports going to either tile or bool for comparisons

Reply:
Ok, then I think I should add those cases. Feels like the responsible thing to do. Although these will never account for `TileResult` - mixing RDD and `LazyTile` would be weird.

Reply:
`classify` I'm going to leave as only handling RDDs. Numbers don't make sense there as input.

Reply:
They might, though - consider landcover classification: I could hand you a raster to classify or else a cell's value for classification under whatever scheme I'm working with